dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/LogProgramStatistics.java

```diff
@@ -0,0 +1,61 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.event.core.*;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class LogProgramStatistics implements ProgramProcessor {
+
+    private static final Logger logger = LogManager.getLogger(LogProgramStatistics.class);
+
+    private LogProgramStatistics() { }
+
+    public static LogProgramStatistics newInstance() { return new LogProgramStatistics(); }
+
+    @Override
+    public void run(Program program) {
+        if (!logger.isInfoEnabled()) {
+            return;
+        }
+
+        int totalEventCount = 0;
+        int storeCount = 0;
+        int loadCount = 0;
+        int initCount = 0;
+        int fenceCount = 0;
+        for (Event e : program.getEvents()) {
+            totalEventCount++;
+            if (e instanceof Store) {
+                storeCount++;
```

Comment:
Here we count `Init` events twice, right? Do you really want that?

Reply:
`Init` events are no `Store` events. Nothing is counted twice.